File: src/main/java/com/aliiensmp/aliienResize/Economy/currencyOptions/RoyaleEco.java
package com.aliiensmp.aliienResize.Economy.currencyOptions;

import com.aliiensmp.aliienResize.Economy.CurrencyProvider;
import me.qKing12.RoyaleEconomy.RoyaleEconomy;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;

public class RoyaleEco implements CurrencyProvider {

    private final String suffix;

    /**
     * Creates the RoyaleEconomy adapter.
     *
     * @param suffix suffix displayed beside prices
     */
    public RoyaleEco(String suffix) {
        this.suffix = suffix;
    }

    @Override
    public boolean isValid() {
        return Bukkit.getPluginManager().getPlugin("RoyaleEconomy") != null;
    }

    @Override
    public boolean hasBalance(Player player, double amount) {
        if (!isValid()) return false;
        return RoyaleEconomy.apiHandler.balance.getBalance(player.getUniqueId().toString()) >= amount;
    }

    @Override
    public boolean withdraw(Player player, double amount) {
        if (!hasBalance(player, amount)) return false;

        RoyaleEconomy.apiHandler.balance.removeBalance(player.getUniqueId().toString(), amount);
        return true;
    }

    @Override
    public String suffix() {
        return suffix;
    }
}

File: src/main/kotlin/com/aliiensmp/aliienResize/Economy/currencyOptions/RoyaleEco.kt
package com.aliiensmp.aliienResize.Economy.currencyOptions

import com.aliiensmp.aliienResize.Economy.CurrencyProvider
import me.qKing12.RoyaleEconomy.RoyaleEconomy
import org.bukkit.Bukkit
import org.bukkit.entity.Player

class RoyaleEco(override val suffix: String) : CurrencyProvider {

    override val isValid: Boolean
        get() = Bukkit.getPluginManager().getPlugin("RoyaleEconomy") != null

    override fun hasBalance(player: Player, amount: Double): Boolean {
        if (!isValid) return false
        return RoyaleEconomy.apiHandler.balance.getBalance(player.uniqueId.toString()) >= amount
    }

    override fun withdraw(player: Player, amount: Double): Boolean {
        if (!hasBalance(player, amount)) return false
        RoyaleEconomy.apiHandler.balance.removeBalance(player.uniqueId.toString(), amount)
        return true
    }
}